File: slideholder/src/main/java/zhan/library/slide/SlideAnimatorListener.java
package zhan.library.slide;

import android.animation.Animator;

public abstract class SlideAnimatorListener implements Animator.AnimatorListener {

    @Override
    public void onAnimationStart(Animator animation) {

    }

    @Override
    public void onAnimationEnd(Animator animation) {

    }

    @Override
    public void onAnimationCancel(Animator animation) {

    }

    @Override
    public void onAnimationRepeat(Animator animation) {

    }

    public abstract void onSlideAnimationStart(Animator animation, int currentStatus);

    public abstract void onSlideAnimationEnd(Animator animation, int currentStatus);
}

File: slideholder/src/main/java/zhan/library/slide/SlideAnimatorListener.kt
package zhan.library.slide

import android.animation.Animator

abstract class SlideAnimatorListener : Animator.AnimatorListener {

    override fun onAnimationStart(animation: Animator) {

    }

    override fun onAnimationEnd(animation: Animator) {

    }

    override fun onAnimationCancel(animation: Animator) {

    }

    override fun onAnimationRepeat(animation: Animator) {

    }

    abstract fun onSlideAnimationStart(animation: Animator, currentStatus: Int)

    abstract fun onSlideAnimationEnd(animation: Animator, currentStatus: Int)
}
